argus/client/base.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass
+from uuid import UUID
+
+import requests
+
+from argus.backend.util.enums import TestStatus
+
+
+class ArgusClient:
+    schema_version: str | None = None
+
+    @dataclass
+    class Routes():
+        SUBMIT = "/testrun/$type/submit"
+        HEARTBEAT = "/testrun/$type/$id/heartbeat"
+        SET_STATUS = "/testrun/$type/$id/set_status"
+        SET_PRODUCT_VERSION = "/testrun/$type/$id/update_product_version"
+        SUBMIT_LOGS = "/testrun/$type/$id/logs/submit"
+        FINALIZE = "/testrun/$type/$id/finalize"
+
+    def __init__(self, auth_token: str, base_url: str, api_version="v1") -> None:
+        self._auth_token = auth_token
+        self._base_url = base_url
+        self._api_ver = api_version
+
+    @property
+    def auth_token(self) -> str:
+        return self._auth_token
+
+    def get_url_for_endpoint(self, endpoint: 'ArgusClient.Routes', location_params: dict[str, str] | None) -> str:
+        if location_params:
+            for param, value in location_params.items():
+                endpoint = endpoint.replace(f"${param}", value)
```

Comment:
add verification that all required params has been replaced (point to missing elements)

Reply:
Done